flink-end-to-end-tests/flink-tpcds-test/src/main/java/org/apache/flink/table/tpcds/utils/AnswerFormatter.java

```diff
@@ -0,0 +1,206 @@
+			String file = ORIGIN_ANSWER_FILE.get(i);
+			String originFileName = file + ANSWER_FILE_SUFFIX;
+			String destFileName = file.split("_")[0] + ANSWER_FILE_SUFFIX;
+			File originFIle = new File(originDir + "/" + originFileName);
+			File destFile = new File(destDir + "/" + destFileName);
+			convert(originFIle, destFile);
+		}
+	}
+
+	/**
+	 * TPC-DS answer set has three kind of formats, recognize them and convert to unified format.
+	 *<p>format 1:
+	 * CUSTOMER_ID     |CUSTOMER_FIRST_NAME |CUSTOMER_LAST_NAME            |C
+	 * ----------------|--------------------|------------------------------|-
+	 * AAAAAAAAAFGBBAAA|Howard              |Major                         |Y
+	 *</p>
+	 * <p>format 2:
+	 * CUSTOMER_ID      CUSTOMER_FIRST_NAME  CUSTOMER_LAST_NAME             C
+	 * ---------------- -------------------- ------------------------------ -
+	 * AAAAAAAAAFGBBAAA Howard               Major                          Y
+	 * </p>
+	 * <p>format 3:
+	 * CUSTOMER_ID|CUSTOMER_FIRST_NAME|CUSTOMER_LAST_NAME|C
+	 * AAAAAAAAAFGBBAAA|Howard|Major|Y
+	 *</p>
+	 * @param originFile
+	 * @param destFile
+	 * @throws Exception
+	 */
+	private static void convert(File originFile, File destFile) throws Exception {
+		BufferedReader reader = new BufferedReader(new FileReader(originFile));
+		BufferedWriter writer = new BufferedWriter(new FileWriter(destFile));
+
+		String line;
+		List<Integer> colLengthList;
+		List<String> content = new ArrayList<>();
+		while ((line = reader.readLine()) != null) {
+			content.add(line);
+		}
+
+		if (isFormat1(content)) {
+			colLengthList = Arrays.stream(content.get(1).split(REGEX_SPLIT_BAR))
+				.map(col -> col.length())
+				.collect(Collectors.toList());
+			writeContent(reader, writer, content, colLengthList);
+		} else if (isFormat2(content)) {
+			colLengthList = Arrays.stream(content.get(1).split(RESULT_HEAD_STRING_SPACE))
+				.map(col -> col.length())
+				.collect(Collectors.toList());
+			writeContent(reader, writer, content, colLengthList);
+		} else {
+			writeContent(reader, writer, content, null);
+		}
+	}
+
+	private static Boolean isFormat1(List<String> content) {
+		if (content.size() > 1 && content.get(1).contains(RESULT_HEAD_STRING_DASH)) {
+			return content.get(0).contains(RESULT_HEAD_STRING_BAR);
+		}
+		return false;
+	}
+
+	private static Boolean isFormat2(List<String> content) {
+		if (content.size() > 1 && content.get(1).contains(RESULT_HEAD_STRING_DASH)) {
+			return true;
+		}
+		return false;
+	}
+
+	private static String formatRow(String row, List<Integer> colLengthList) {
+		StringBuilder sb = new StringBuilder();
+		int start;
+		int end = 0;
+		for (int i = 0; i < colLengthList.size(); i++) {
+			if (i == 0) {
+				start = end;
+			} else {
+				start = end + SPACE_BETWEEN_COL;
+			}
+			end = start + colLengthList.get(i);
+			end = end < row.length() ? end : row.length();
+			try {
+				sb.append(row.substring(start, end).trim());
+			} catch (Exception e) {
```

Comment:
why catch exception here?

Reply:
start may larger than end for some case, I'll rewrite this piece.